LiteCore/Query/SQLiteDataFile+Indexes.cc

```diff
@@ -259,7 +275,10 @@ namespace litecore {
                 auto what = spec.what();
                 // `what()` is defined as an array of 1+ exprs to index; for a vector index there can be only one.
                 // In some cases just that term is passed in, not wrapped in an array.
-                if ( what->count() > 1 || what->get(0)->type() == kArray ) what = (const Array*)what->get(0);
+                if ( what->count() > 1
+                     || (spec.queryLanguage == QueryLanguage::kN1QL || what->get(0)->type() == kArray) ) {
```

Comment:
Is this change needed because what->get(0) may not be an array when using lazy index?

Reply:
yes